### PR TITLE
Add fortran openmp sample

### PR DESCRIPTION
**Pull request by krisdale**
# Adding a New Sample(s)

## Description

This PR adds a new sample to the DirectProgramming folder for the Fortran Compiler. Sample shows how to use OpenMP offload for GPU, using a matrix multiplication program as the example.

## Checklist
Administrative
- [X] Review sample design with the appropriate [Domain Expert](https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts): Barbara Perz

Code Development
- [ ] Implement coding guidelines and ensure code quality. [see wiki for details](https://github.com/oneapi-src/oneAPI-samples/wiki/General-Code-Guidelines)
- [X] Adhere to readme template 
- [ ] Enforce format via clang-format config file
- [X] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [ ] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [X] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com

Security and Legal
- [ ] OSPDT Approval (see Project Manager for assistance)
- [ ] Compile using the following compiler flags and fix any warnings, the falgs are: "/Wall -Wformat-security -Werror=format-security"
- [ ] Bandit Scans (Python only)
- [ ] Virus scan

Review
- [ ] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [ ] Review readme with Tom Lenth(@tomlenth) and/or Project Manager
- [ ] Tested using Dev Cloud when applicable

### CODE REVIEW

**Review by jkinsky — APPROVED** [review]

**Comment by jimmytwei** [comment]
@krisdale Would you like me to merge PR?

File: DirectProgramming/Fortran/guided_matrix_mul_OpenMP/README.md [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+## Build and Run the Parallel Program
+
+In order for the Intel Fortran Compiler to recognize OpenMP directives, the program must be compiled with the `-qopenmp` (Linux) or `/Qopenmp` (Windows) option. Without this compiler option, the OpenMP directives are ignored and the program will run sequentially. Note that this can be helpful for debugging the program.
+
+### On Linux
+
+1. Change to the sample directory.
+2. Build the program using the `-qopenmp` option.
+   ```
+   ifx -xhost -qopenmp mm_CPUparallel.f90
```
Comment (jimmytwei) [inline DirectProgramming/Fortran/guided_matrix_mul_OpenMP/README.md:175]: @krisdale I think you need to change this from `mm_CPUparallel.f90` to `02_mm_CPU_parallel.f90`

File: DirectProgramming/Fortran/guided_matrix_mul_OpenMP/README.md [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+In order for the Intel Fortran Compiler to recognize OpenMP directives, the program must be compiled with the `-qopenmp` (Linux) or `/Qopenmp` (Windows) option. Without this compiler option, the OpenMP directives are ignored and the program will run sequentially. Note that this can be helpful for debugging the program.
+
+To compile for the device, use the `-fopenmp-targets=spir64` (Linux) or `/Qopenmp-targets:spir64` (Windows) option. This option enables the compiler to generate a x86 + SPIR64 fat binary for the device binary generation. “SPIR64” stands for "64-bit Standard, Portable Intermediate Representation".
+
+### On Linux
+
+1. Change to the sample directory.
+2. Build the program using the `-qopenmp`  and `-fopenmp-targets=spir64` options.
+   ```
+   ifx -xhost -qopenmp -fopenmp-targets=spir64 mm_GPU.f90
```
Comment (jimmytwei) [inline DirectProgramming/Fortran/guided_matrix_mul_OpenMP/README.md:286]: @krisdale I think you need to change `mm_GPU.f90` to `03_mm_GPU.f90`

File: DirectProgramming/Fortran/guided_matrix_mul_OpenMP/README.md [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+   make gpu
+   ```
+
+### On Windows
+
+1. Open an Intel oneAPI command window.
+2. Change to the sample directory.
+3. Build the program using the `/Qopenmp` option.
+   ```
+   ifx /Qxhost /Qopenmp /Qopenmp-targets:spir64 mm_GPU.f90
```
Comment (jimmytwei) [inline DirectProgramming/Fortran/guided_matrix_mul_OpenMP/README.md:311]: @krisdale I think you need to change `mm_GPU.f90` to `03_mm_GPU.f90`

**Comment by jimmytwei** [comment]
@krisdale Still having issues with the Makefile getting this compiled.

**Review by IgorOchocki — APPROVED** [review]

**Comment by krisdale** [comment]
@IgorOchocki @jimmytwei  - Are there still issues with the Makefile (per @jimmytwei 's last comment)? If so, what is the issue that you encounter? Or is this PR approved for merge?

**Comment by jimmytwei** [comment]
@krisdale I'm still getting errors running `make seq` , `make par` and make `gpu`. I think the Makefile needs to be updated with the correct filenames.

I've also made comments in the README.md file - the filenames need to be updated here as well.

**Review by jimmytwei — APPROVED** [review]

**Comment by jimmytwei** [comment]
@krisdale LGTM. The code compiles using the provided Makefile. I made some changes to the README to make sure that the filenames are consistent with the files in the src folder. Merging this PR.

**Pull request merged**